src/Compilers/CSharp/Test/Syntax/Parsing/ParsingErrorRecoveryTests.cs

```diff
@@ -4925,14 +4925,43 @@
             Assert.NotEqual(SyntaxKind.None, ds.Declaration.Variables[0].Initializer.EqualsToken.Kind());
             Assert.NotNull(ds.Declaration.Variables[0].Initializer.Value);
             Assert.Equal(SyntaxKind.ObjectCreationExpression, ds.Declaration.Variables[0].Initializer.Value.Kind());
-            Assert.Equal(1, file.Errors().Length);
-            Assert.Equal((int)ErrorCode.ERR_RbraceExpected, file.Errors()[0].Code);
+            var errors = file.Errors();
+            Assert.Equal(3, errors.Length);
```

Comment:
>Assert.Equal(3, errors.Length); [](http://example.com/codeflow?start=12&length=31)

We appear to be reporting more errors in several of these cases. Is that expected?

Reply:
I do consider this expected. Before the `;` was interpreted as the end of the initializer expression (as if a `}` was written). With these changes, if a `}` or expression follows the semicolon, the `;` will not terminate the initializer expression. 

As a result more accurate errors are generated (imo). In this test case for example the 'new' errors are 1. a syntax error as a `;` is not expected in an initializer, 2. a missing `;` after the next `}` (which terminates the initializer expression) and 3. the missing `}` to close the outer-most scope.